### PR TITLE
Feat: Beta CSR compatibility

### PR DESCRIPTION
**Pull request by yue9944882**

### CODE REVIEW

**Review by qiujian16 — COMMENTED** [review]
/approve

/assign @xuezhaojun

File: go.mod [hunk 1]
```diff
@@ -108,3 +108,5 @@ require (
 )
 
 replace sigs.k8s.io/apiserver-network-proxy/konnectivity-client => sigs.k8s.io/apiserver-network-proxy/konnectivity-client v0.0.27
+
+replace open-cluster-management.io/addon-framework => github.com/yue9944882/addon-framework v0.0.0-20220315104416-d9192aa14201
```
Comment (xuezhaojun) [inline go.mod:112]: I guess it should depend on the addon-framework repo under ocm.

File: pkg/util/csr.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import (
+	"github.com/pkg/errors"
+	certificatesv1 "k8s.io/api/certificates/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/discovery/cached/memory"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/restmapper"
+)
+
+func IsCSRV1Supported(nativeClient kubernetes.Interface) (bool, error) {
```
[the remark this answers is not in the document]
Comment (yue9944882) [reply pkg/util/csr.go]: this is the removed in the latest revision

File: go.mod (hunk 1 above)
Comment (yue9944882) [reply go.mod]: updated to latest commit of addon-framework

File: pkg/proxyagent/agent/manifests/charts/addon-agent/templates/agent-client-secret.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{{ if .Values.includeStaticProxyAgentSecret }}
+apiVersion: v1
+kind: Secret
+metadata:
+  namespace: {{ .Release.Namespace }}
+  name: cluster-proxy-open-cluster-management.io-proxy-agent-signer-client-cert
+data:
+  "tls.crt": {{ .Values.staticProxyAgentSecretCert }}
+  "tls.key": {{ .Values.staticProxyAgentSecretKey }}
+{{ end }}
```
[the remark this answers is not in the document]
Comment (yue9944882) [reply pkg/proxyagent/agent/manifests/charts/addon-agent/templates/agent-client-secret.yaml]: added

File: pkg/proxyagent/agent/agent.go [hunk 4]
```diff
@@ -160,6 +174,17 @@ func GetClusterProxyValueFunc(runtimeClient client.Client, nativeClient kubernet
 		if serviceEntryPointPort == 0 {
 			serviceEntryPointPort = 8091
 		}
+		certDataBase64, keyDataBase64 := "", ""
+		if !v1CSRSupported {
```
[the remark this answers is not in the document]
Comment (yue9944882) [reply pkg/proxyagent/agent/agent.go:182]: i can't think of a better name for now, can you give me some hints?

File: pkg/util/csr.go [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+	v1beta1CSRSupported := false
+	for _, mapping := range mappings {
+		if mapping.GroupVersionKind.Version == "v1beta1" {
+			v1beta1CSRSupported = true
+		}
+	}
+	if !v1beta1CSRSupported {
+		return false, errors.New("No available CSR API in the hub cluster")
+	}
+	return false, nil
```
[the remark this answers is not in the document]
Comment (xuezhaojun) [reply pkg/util/csr.go]: Oh, I see the  IsCSRSupported return two variables, it's fine now.

File: pkg/proxyagent/agent/agent.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (xuezhaojun) [reply pkg/proxyagent/agent/agent.go:182]: Never mind, with comments it seems clear to me.

**Pull request merged**